### ISSUE TITLE
Converter not being called on all server responses

### ISSUE THREAD

**Issue by danvinokour** [square/retrofit#1564]
Our server passes messages if we get unauthorised code and so on...
Can you call the converterFactory on all return codes and not only on 200?

**Comment by JakeWharton** [comment]
No, as a proxy, load balancer, or the HTTP server may trigger non-200 status codes with bodies in an unexpected format (like HTML).

This sample shows how to deserialize error bodies: https://github.com/square/retrofit/blob/3c195a0cc0c95b82b59bf3f88f6eabed3eeb3621/samples/src/main/java/com/example/retrofit/DeserializeErrorBody.java#L60-L69

**Comment by danvinokour** [comment]
Its a little bit messy. Isn't it better to add an error converter beforehand?

**Comment by JakeWharton** [comment]
Before what? And add where?

**Comment by danvinokour** [comment]
I meant either to make it as the regular ConverterFactory or by adding ErrorConverterFactory when creating the retrofit object.

Why converting the error should be harder and different then converting the regular body?

**Comment by JakeWharton** [comment]
I listed reasons why above. You cannot guarantee the format of the response
in the non-success case.

**Comment by danvinokour** [comment]
In your example you say that its my responsibility to know whether certain code comes with the body im expecting.
After i talked with our server guys they say that i cannot truly know if a certain code with a format that i expect.
They said that it is best to check the Content-Type header to understand at least that its a Json.

So what i get is if i use your example as its described i can still crash there or fail becouse i can never assume that the error was generated by server as i expect.

In conclusion, maybe you could add a support for error convertors by content-type?
